### PR TITLE
Fix Range op: support negative delta and add validations

### LINKED ISSUE #235: Range op: negative delta wraps to huge usize, causing incorrect output
## Summary

In the Range op codegen, `delta` is cast with `as usize` for `Tensor::arange_step()`. Since `RangeInput` values are `i64` and the ONNX spec allows negative `delta` (for descending ranges), a negative runtime/static value will wrap to a huge `usize`, leading to incorrect output or very large allocations.

## Location

- Codegen: `crates/burn-onnx/src/burn/node/range.rs:53`
- IR TODOs already noting missing coverage: `crates/onnx-ir/src/node/range.rs:134-149`

## Suggested fix

Either:
1. Validate `delta > 0` (and `delta != 0`) during IR processing in `extract_config` or `infer_types`, rejecting unsupported cases with a `ProcessError`
2. Or generate code that correctly supports descending ranges (negative delta)

Option 1 is simpler and sufficient until Burn's `arange_step` supports descending ranges natively.

## Related TODOs in range.rs

- `delta=0` edge case (division by zero)
- Negative delta (descending range) test coverage
- Empty range validation

### PR DESCRIPTION
**Pull request by antimora**
## Summary

Fixes #235

- Replace `arange_step(start..limit, delta as usize)` with formula-based approach `arange(0..n) * delta + start` where `n = max(ceil((limit - start) / delta), 0)`, correctly handling positive, negative, and empty ranges
- Add dtype validation (all 3 inputs must match), scalar validation (rank <= 1), and static delta=0 rejection in onnx-ir
- Add end-to-end tests for negative delta (static + runtime) and empty ranges

## Test plan

- [x] Codegen snapshot tests: static, negative delta, empty range
- [x] onnx-ir unit tests: dtype mismatch, non-scalar input, delta=0
- [x] Integration tests: static negative delta model, runtime negative delta, empty range cases
- [x] All existing range tests still pass (static, mixed, runtime)

### CODE REVIEW

File: crates/burn-onnx/src/burn/node/range.rs [hunk 1]
```diff
@@ -41,17 +41,45 @@ impl NodeCodegen for onnx_ir::node::range::RangeNode {
...
+                        let __delta = #delta;
+                        let __n = ((__limit - __start) as f64 / __delta as f64)
+                            .ceil().max(0.0) as i64;
+                        Tensor::arange(0..__n, &*self.device)
+                            .cast(#output_dtype)
+                            .mul_scalar(__delta)
+                            .add_scalar(__start)
+                    };
+                }
+            }
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-onnx/src/burn/node/range.rs:83]: Already addressed in 8b33d767 — added `test_range_runtime` snapshot test covering the `_ =>` arm with all three `RangeInput::Runtime` parameters.
